### ISSUE TITLE
Auto-Type window filtering has a delay since 2.7.0

### ISSUE THREAD

**Issue by pauschuu** [keepassxreboot/keepassxc#7596]
## Overview
Before 2.7.0 I used to press my Auto-Type shortcut, which opened the Auto-Type window. To select a common entry I just pressed one letter on the keyboard to have that entry selected/filtered in the Auto-Type window. I immediately pressed enter after pressing the key to filter the values and it worked well. Now when typing (in order to filter) there is a delay.

## Steps to Reproduce
1. Have multiple Entries that would show up on a window of your choice when pressing the Auto-Type shortcut
2. Press the Auto-Type shortcut in the window of your choice
3. Type letters to filter the entries in the Auto-Type window
4. See the delay
5. You can't press immediately Enter after typing your filter letter/string because of the delay you would select the wrong entry

## Expected Behavior
I can just do this in the fastest possible way without a delay that causes me to select the wrong entry if I am too fast:
1. Press Auto-Type shortcut
2. Press keys to filter the Auto-Type entries
3. Press Enter

## Actual Behavior
I have to wait when doing the step 2 in "Expected behaviour" because of the delay.

## Context
came with 2.7.0

```
KeePassXC - Version 2.7.0
Revision: d7a9ef4

Qt 5.15.3
Debugging mode is disabled.

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 5.16.16-arch1-1

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.19.1
```

Operating System: Linux
Desktop Env: [dwm](https://dwm.suckless.org/)
Windowing System: X11

**Comment by droidmonkey** [comment]
The search timeout is 0.3 seconds. This is important when searching the entire database, but not when searching the filtered list. Could remove the delay when not searching the database.

**Comment by pauschuu** [comment]
Oh and in addition to that:
After waiting till the delay is done, the filtered entry that pops up (only entry in my case), oftentimes isn't selected. Before 2.7.0 I'd say it was always selected. (Not selected meaning I also have to press an arrow down key before pressing enter)

**Comment by droidmonkey** [comment]
I cannot reproduce the last behavior you mentioned. In either case you should just press enter to type it.

**Comment by pauschuu** [comment]
> I cannot reproduce the last behavior you mentioned. In either case you should just press enter to type it.

Doesn't occur with your fix from #7598. 
So consider this done, thx. :)
